add pkgs.darwin.apple_sdk.frameworks.Security to flake.nix

rjpcasalino opened the pull request:
for some reason in order for `cargo test` to work on Darwin I needed `pkgs.darwin.apple_sdk.frameworks.Security`. Is there a better way to do this?

06kellyjac commented:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/gemini/agate/default.nix#L16

`lib.optionals` and `stdenv.isDarwin` are generally go-tos

In this case you would be doing   `++ lib.optionals stdenv.isDarwin [ Security ]` (you can pull `Security` straight out of `pkgs`)

rjpcasalino committed:
add pkgs.darwin.apple_sdk.frameworks.Security to flake.nix

for some reason in order for `cargo test` to work on Darwin I needed `pkgs.darwin.apple_sdk.frameworks.Security`. Is there a better way to do this?

update flake.nix to inherit darwin framework(s)

06kellyjac commented:
Oh yeah, this is why agate etc can pull Security directly, https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L160

jonringer commented:
`>.>` Not just 1 but many people are looking at this repo. I should really try to finish this... after the release though :(

Anyway, I don't use darwin, but this makes sense that it's needed, as openssl was also need.

jonringer reviewed (approved):
Makes sense. Thanks :)
